src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_PurchaseTabScreen/use-recent-fare-contracts.ts

```diff
@@ -125,17 +125,33 @@ const mapBackendRecentFareContracts = (
     recentFareContract.zones.slice(-1)[0],
   );
 
+  const pointToPointValidity = recentFareContract.point_to_point_validity;
+
+  const orderId = recentFareContract.order_id;
+
+  const fromId =
+    pointToPointValidity?.fromPlace !== undefined
```

Comment:
Maybe `fromId = pointToPointValidity?.fromPlace || fromTariffZone?.id`. Or is there a reason for the explicit check on `undefined`?